File: src/Talker_Listener/talker.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include <sstream>

int main(int argc, char **argv)
{

    /* ----- NODE ----- */
    // Create and initialize node
    ros::init(argc, argv, "talker_cpp");

    // Create node handle
    ros::NodeHandle nodeHandle; // EXTRA

    /* ----- PUBLISHER ----- */
    // Create publisher
    // State type; arg: node_name, queue_size
    ros::Publisher publisher = nodeHandle.advertise<std_msgs::String>("test_talk", 10);

    // Define publication rate
    ros::Rate rate(1);

    int i = 0;
    while (ros::ok())
    {
        std::stringstream ss;
        ss << "Count: " << i;
        std_msgs::String message;
        message.data = ss.str();

        // Put string to log
        ROS_INFO("CPP - %s\n", message.data.c_str());

        // Publish message
        publisher.publish(message);
        ros::spinOnce(); // EXTRA

        // Control message sending rate
        rate.sleep();

        i++;
    }

    return 0;
}

File: src/Talker_Listener/talker.py
import rospy
from std_msgs.msg import String

def talker():
    ## ----- NODE ----- ##
    # Create and initialize node
    # arg: node_name, anonymous (create different ID for different nodes)
    rospy.init_node('talker_py', anonymous=True)

    ## ----- PUBLISHER ----- ##
    # Create publisher object 
    # arg: topic_name, topic_type, queue_size
    publisher = rospy.Publisher('test_talk', String, queue_size=10)

    # Define publication rate (Hz)
    rate = rospy.Rate(1) 

    # Create message
    i = 0
    while not rospy.is_shutdown():
        msg_str = "Counter: %s" % i
        # Put string to log
        rospy.loginfo("Python - %s" % msg_str)
        # Publish message
        publisher.publish(msg_str)
        # Control message sending rate
        rate.sleep()
        
        i+=1

if __name__ == "__main__":
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

